File: 1346A/main.py
t = int(input())
for _ in range(t):
    n, k = map(int, input().split())
    x = n / (1 + k + k ** 2 + k ** 3)
    print(f"{int(x)} {int(x * k)} {int(x * k ** 2)} {int(x * k ** 3)}")

File: 1346A/main.kt
fun main() {
    val t = readLine()!!.toInt()
    repeat(t) {
        val (n, k) = readLine()!!.split(" ").map { it.toInt() }
        val x = n.toDouble() / (1 + k + k * k + k * k * k)
        println("${x.toInt()} ${(x * k).toInt()} ${(x * k * k).toInt()} ${(x * k * k * k).toInt()}")
    }
}